plugins/dell-dock/fu-dell-dock-common.c

```diff
@@ -71,8 +71,8 @@ fu_dell_dock_clone_updatable(FuDevice *device)
 		fu_device_add_flag(device, FWUPD_DEVICE_FLAG_UPDATABLE);
 	} else {
 		const gchar *message = fu_device_get_update_error(parent);
-		if (message != NULL)
-			fu_device_set_update_error(device, message);
-		fu_device_remove_flag(device, FWUPD_DEVICE_FLAG_UPDATABLE);
+		if (message == NULL)
+			message = "Parent is not updatable";
+		fu_device_inhibit(device, "parent-not-updatable", message);
```

Comment:
@CragW needs to review this more closely.

Reply:
Hang on one minute; we have a flag to do all this automatically! `FU_DEVICE_INTERNAL_FLAG_INHIBIT_CHILDREN`

Reply:
It's not just inhibiting children that matters but also setting up all the right ones to gain the updatable flag. If it's the inhibition route then I think that flag needs to be added by default to everything.

Reply:
Ahh, of course, because it's adding it too. Will see how it looks.

Reply:
It looks much better IMO. @CragW if you could test this PR on actual hardware I think we'd all feel more warm and fuzzy. Thanks!

Reply:
I've merged the uncontroversial parts to main, and the others we can test and merge post release. There are a few vendors who are nagging me for a tarball now.